cli/waiter/querying.py

```diff
@@ -67,3 +67,31 @@ def submit(cluster, executor):
         return executor.submit(get_token_on_cluster, cluster, token)
 
     return query_across_clusters(clusters, submit)
+
+
+def services_using_token(cluster, token_name):
+    """Retrieves all services that are using the token"""
+    params = {'token': token_name}
+    services, _ = http_util.make_data_request(cluster, lambda: http_util.get(cluster, 'apps', params=params))
+    return services
+
+
+def get_services_on_cluster(cluster, token_name):
+    """Gets the service(s) using the given token name on the given cluster"""
+    services = services_using_token(cluster, token_name)
+    if services:
+        return {'count': len(services), 'services': services}
+    else:
+        return {'count': 0}
+
+
+def query_services(clusters, token_name):
+    """
+    Uses query_across_clusters to make the service
+    requests in parallel across the given clusters
+    """
+
+    def submit(cluster, executor):
```

Comment:
Can we name this something else? Maybe `query_services_on_cluster `

Reply:
I made this a `lambda` instead of a `def`